main.tf

```diff
@@ -451,13 +460,66 @@ resource "azurerm_virtual_network" "agentless_orchestrate" {
   resource_group_name = local.scanning_resource_group_name
   address_space       = ["10.0.0.0/16"]
 
-  subnet {
-    name           = length(var.custom_network) > 0 ? "" : lower(replace("${local.prefix}-subnet-${local.suffix}-${local.region}", " ", ""))
-    address_prefix = "10.0.0.0/16"
-    security_group = azurerm_network_security_group.agentless_orchestrate[0].id
-  }
+  tags = var.tags
+}
+
+resource "azurerm_subnet" "agentless_subnet" {
+  depends_on = [azurerm_resource_group.scanning_rg]
+  count      = var.regional && length(var.custom_network) == 0 ? 1 : 0
+
+  name                 = lower(replace("${local.prefix}-subnet-${local.suffix}-${local.region}", " ", ""))
+  resource_group_name  = local.scanning_resource_group_name
+  virtual_network_name = azurerm_virtual_network.agentless_orchestrate[0].name
+  address_prefixes     = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet_network_security_group_association" "agentless_nsg_association" {
+  count = var.regional && length(var.custom_network) == 0 ? 1 : 0
+
+  subnet_id                 = azurerm_subnet.agentless_subnet[0].id
+  network_security_group_id = length(var.custom_network_security_group) > 0 ? var.custom_network_security_group: azurerm_network_security_group.agentless_orchestrate[0].id
 }
 
+resource "azurerm_public_ip" "agentless_public_ip" {
+  depends_on = [azurerm_resource_group.scanning_rg]
+  count = var.regional && var.use_nat_gateway ? 1 : 0
+
+  name                = replace("${local.prefix}-public-ip-${local.region}-${local.suffix}", " ", "-")
+  location            = local.region
+  resource_group_name = local.scanning_resource_group_name
+  allocation_method   = "Static"
```

Comment:
I see some thought has gone into deciding the sku and allocation method. The current choice seems to be the right fit.

Reply:
Yeah, the allocation method was just a choice on my part. The SKU actually needs to be set, otherwise it defaults to `Basic` which is incompatible with some of the other resources we deploy. (This is actually contrary to the `azurerm` docs, which indicate that `Standard` is the default. But I digress...)